### PR TITLE
Add example on volume share slippage model

### PR DESCRIPTION
**Pull request by LouisSzeto**
#### Description
Add example on `VolumeShareSlippageModel` to demonstrate the usage of the slippage model.

#### Motivation and Context
No example yet.

#### Requires Documentation Change
Add link reference in Writing Algorithms / Reality Modeling / Slippage / Key Concepts

#### How Has This Been Tested?
Regression test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [X] I have added tests to cover my changes. 
- [X] All new and existing tests passed.
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Comment by LouisSzeto** [comment]
JB comment: Add python version of the model.

File: Algorithm.CSharp/VolumeShareSlippageModelAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
+
```
Comment (jhonabreul) [inline Algorithm.CSharp/VolumeShareSlippageModelAlgorithm.cs:1]: nit - blank line

File: Algorithm.Python/VolumeShareSlippageModelAlgorithm.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+### </summary>
+class VolumeShareSlippageModelAlgorithm(QCAlgorithm):
+    longs = []
+    shorts = []
+
+    def initialize(self) -> None:
+        self.set_start_date(2021, 1, 1)
+        self.set_end_date(2022, 1, 1)
+        # To set the slippage model to limit to fill only 30% volume of the historical volume, with 5% slippage impact.
+        self.set_security_initializer(lambda security: security.set_slippage_model(VolumeShareSlippageModel(0.3, 0.05)))
```
Comment (jhonabreul) [inline Algorithm.Python/VolumeShareSlippageModelAlgorithm.py:27]: Should import the added Python `VolumeShareSlippageModel` class. This is using C#'s one, so the Python one is untested

**Review by jhonabreul — COMMENTED** [review]
Leaving another comment

File: Algorithm.CSharp/VolumeShareSlippageModelAlgorithm.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            // Create QQQ symbol to explore its constituents.
+            var qqq = QuantConnect.Symbol.Create("QQQ", SecurityType.Equity, Market.USA);
+
+            // Weekly updating the portfolio to allow time to capitalize from the popularity gap.
+            UniverseSettings.Schedule.On(DateRules.WeekStart());
+            // Add universe to trade on the most and least liquid stocks among QQQ constituents.
+            AddUniverse(
+                // First we select from all QQQ constituents to the next filter on liquidity.
+                Universe.ETF(qqq.Value, Market.USA, UniverseSettings, (constituents) => constituents.Select(c => c.Symbol)),
```
[the remark this answers is not in the document]
Comment (LouisSzeto) [reply Algorithm.CSharp/VolumeShareSlippageModelAlgorithm.cs:50]: I tried but there will be an error of `Symbol should not contains "|" or " "`. It is fine if I add QQQ by `AddEquity`. Maybe it is a Lean's issue on `Symbol.Create`?

**Review by Martin-Molinero — APPROVED** [review]
Thank you 👍

**Pull request merged**